server/grpc/rpc_test.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2022 Woodpecker Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grpc
+
+import (
+	"context"
+	"testing"
+
+	"github.com/franela/goblin"
+	"github.com/stretchr/testify/assert"
+	"go.woodpecker-ci.org/woodpecker/v2/server/model"
+	mocks_store "go.woodpecker-ci.org/woodpecker/v2/server/store/mocks"
+	"google.golang.org/grpc/metadata"
+)
+
+func TestRegisterAgent(t *testing.T) {
+	g := goblin.Goblin(t)
+	g.Describe("When existing agent Name is empty", func() {
+		g.It("Should update Name with hostname from metadata", func() {
+			store := mocks_store.NewStore(t)
+			storeAgent := new(model.Agent)
+			storeAgent.ID = 1337
+			updatedAgent := model.Agent{
+				ID:          1337,
+				Created:     0,
+				Updated:     0,
+				Name:        "hostname",
+				OwnerID:     0,
+				Token:       "",
+				LastContact: 0,
+				Platform:    "platform",
+				Backend:     "backend",
+				Capacity:    2,
+				Version:     "version",
+				NoSchedule:  false,
+			}
+
+			store.On("AgentFind", int64(1337)).Once().Return(storeAgent, nil)
+			store.On("AgentUpdate", &updatedAgent).Once().Return(nil)
```

Comment:
I wanted to do something like
```suggestion
			var updatedAgent *model.Agent

			store.On("AgentFind", int64(1337)).Once().Return(storeAgent, nil)
			store.On("AgentUpdate", mock.AnythingOfType("*model.Agent")).Once().Return(func(agent *model.Agent) {
				updatedAgent = agent
			})
```
And then in the end of the test, check 
```
			assert.EqualValues(t, "hostname", updatedAgent.Name)
			assert.EqualValues(t, "platform", updatedAgent.Platform)
			assert.EqualValues(t, "backend", updatedAgent.Backend)
			assert.EqualValues(t, "version", updatedAgent.Version)
```
But I could not get that to work.
It says 
```
panic: assert: arguments: Error(0) failed because object wasn't correct type: 0x906f80
```

Reply:
It is not the `mock.AnythingOfType("*model.Agent")`. I get the same error if I use `mock.Anything`.
Also, for some reason, the same error if I use
```go
			var updatedAgent model.Agent

			store.On("AgentFind", int64(1337)).Once().Return(storeAgent, nil)
			store.On("AgentUpdate", mock.Anything).Once().Return(func(agent model.Agent) {
				updatedAgent = agent
			})
```

Reply:
I am not going to use more effort on improving the tests